### DISCUSSION TITLE
React 18

### DISCUSSION THREAD

**Discussion by anjanesh** [vercel/next.js#33334] [Help] [answered]
Which version of NextJS will include React 18 ?

**Comment by balazsorban44** [comment, marked as answer]
Hi, you can use Next.js with React 18 already!

Please refer to the docs: https://nextjs.org/docs/advanced-features/react-18#react-18-usage-in-nextjs

**Reply by anjanesh** [reply]
Even for production ? If not, how to update from RC to stable in the future ?

**Reply by balazsorban44** [reply]
Please follow the link, you will see there is currently no stable release of React 18 (it is currently a Release Candidate). This is unrelated to Next.js.

When it comes out, Next.js will have full support.

**Reply by anjanesh** [reply]
What I meant was, if I do this now :
`npm install next@latest react@rc react-dom@rc`
When React 18 is released, how do I update my NextJS's React codebase to the stable version of 18 ?

**Reply by balazsorban44** [reply]
`npm install react@latest react-dom@latest`

We will likely have a blog post and updated docs by then.

https://nextjs.org/blog is a good place to follow news about Next.js

**Reply by anjanesh** [reply]
Thank You.

**Reply by raajnadar** [reply]
@balazsorban44 please update the version that supports the stable version of React Js.
